tiledb/sm/serialization/query.cc

```diff
@@ -503,11 +504,6 @@ Status query_from_capnp(
         "Cannot deserialize; Query opened for " + query_type_str(type) +
         " but got serialized type for " + query_reader.getType().cStr()));
 
-  // Deserialize layout.
-  Layout layout = Layout::UNORDERED;
-  RETURN_NOT_OK(layout_enum(query_reader.getLayout().cStr(), &layout));
-  RETURN_NOT_OK(query->set_layout(layout));
-
   // Deserialize array instance.
   RETURN_NOT_OK(array_from_capnp(query_reader.getArray(), array));
```

Comment:
Does this need to come after the overflow check as well?

Reply:
Yeah -- just to adhear to the contract that 'query' is not mutated before the overflow check.

Reply:
I don't see any scenario where this would really matter, since the layout should be consistent between returns. This whole patch will get a hard-revert in the future in favor of the real fix, because I agree it makes more sense conceptually to be placed where it was.

Reply:
I meant the `array_from_capnp()` which is mutating the query's array instance I believe.

Reply:
Oh, I see. You're right, this mutates the query array instance. Is this OK to move lower in the call? I don't see 'array' explicitly referenced anywhere else within query_from_capnp().

Reply:
Yes it should be fine to move lower in the function.